__tests__/integration/melt-tokens.test.js

```diff
@@ -0,0 +1,342 @@
+import { TestUtils, WALLET_CONSTANTS } from './utils/test-utils-integration';
+import { WalletHelper } from './utils/wallet-helper';
+
+describe('melt tokens', () => {
+  let wallet1;
+  const tokenA = {
+    name: 'Token A',
+    symbol: 'TKA',
+    uid: null
+  };
+
+  beforeAll(async () => {
+    wallet1 = new WalletHelper('melt-token-1');
+
+    // Starting the wallets
+    await WalletHelper.startMultipleWalletsForTest([wallet1]);
+
+    // Creating a token for the tests
+    await wallet1.injectFunds(10, 0);
+    const tkAtx = await wallet1.createToken({
+      name: tokenA.name,
+      symbol: tokenA.symbol,
+      amount: 800,
+      address: await wallet1.getAddressAt(0),
+      change_address: await wallet1.getAddressAt(0)
+    });
+    tokenA.uid = tkAtx.hash;
+
+    /**
+     * Status:
+     * wallet1[0]: 2 HTR , 800 TKA
+     */
+  });
+
+  afterAll(async () => {
+    await wallet1.stop();
+  });
+
+  // Testing failures first, that do not cause side-effects on the blockchain
+
+  it('should not melt an invalid token', async done => {
+    const response = await TestUtils.request
+      .post('/wallet/melt-tokens')
+      .send({
+        token: 'invalidToken',
+        amount: 100
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    expect(response.status).toBe(200);
+    expect(response.body.success).toBe(false);
+
+    // Even though the result is correct, the error thrown is not related. Should be fixed later.
```

Comment:
Add `TODO` to the comment so we can find it using grep later

Reply:
Done :+1: